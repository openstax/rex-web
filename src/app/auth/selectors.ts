import { createSelector } from 'reselect';
import * as selectNavigation from '../navigation/selectors';
import * as parentSelectors from '../selectors';

export const localState = createSelector(
  parentSelectors.localState,
  (parentState) => parentState.auth
);

export const user = createSelector(
  localState,
  (state) => state.user
);

export const loggedOut = createSelector(
  localState,
  user,
  (state, currentUser) => state.established && !currentUser
);

export const loginLink = createSelector(
  selectNavigation.pathname,
  (currentPath) => '/accounts/login?r=' + currentPath
);

export const signupLink = createSelector(
  selectNavigation.pathname,
  (currentPath) => '/accounts/signup?r=' + currentPath
);
