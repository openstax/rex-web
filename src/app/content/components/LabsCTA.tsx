import React from 'react';
import { FormattedMessage } from 'react-intl';
import { useSelector } from 'react-redux';
import styled, { css } from 'styled-components/macro';
import icon from '../../../assets/kinetic.svg';
import Button from '../../components/Button';
import { kineticBannerVariant } from '../../featureFlags/selectors';
import theme from '../../theme';
import { disablePrint } from './utils/disablePrint';

// tslint:disable-next-line: variable-name
const LabsLogo = styled.img`
  width: 5.5rem;
`;

// tslint:disable-next-line: variable-name
const LabsCTAWrapper = styled.div`
  width: 100%;
  max-width: 82.5rem;
  margin: 0 auto 1.6rem;
  padding: 2.5rem;
  display: flex;
  align-items: center;
  background-color: #151b2c;
  ${theme.breakpoints.mobileMedium(css`
    padding: 1.6rem;
    align-items: flex-start;
    flex-direction: column;
  `)}
  ${disablePrint}
`;

/* stylelint-disable block-opening-brace-newline-after, block-closing-brace-newline-before */
/* stylelint-disable block-opening-brace-space-after, block-closing-brace-space-before */
// tslint:disable-next-line: variable-name
const Column = styled.div`
  display: flex;
  ${(props) => props.flex ? css`flex: 1;` : null}
  flex-direction: column;
  color: ${theme.color.white};
  overflow: hidden;
  ${(props) => props.maxWidth ? css`max-width: ${props.maxWidth}rem;` : null}
  ${(props) => !props.last ? css`
    margin-right: 2.5rem;
    ${theme.breakpoints.mobileMedium(css`
      margin-right: 0;
      margin-bottom: 3rem;
    `)}
  ` : null}

  ${theme.breakpoints.mobileMedium(css`
    margin-right: 0;
    max-width: unset;
  `)}
`;

// tslint:disable-next-line: variable-name
const LabsCTAHeader = styled.div`
  display: flex;
  flex-direction: row;
  ${theme.breakpoints.mobileMedium(css`
    width:100%;
    align-items: center;

    ${LabsLogo} {
      margin-right: 2.5rem;
    }
  `)}
`;

// tslint:disable-next-line: variable-name
const LabsText = styled.div`
  font-size: ${(props) => props.size}rem;
  font-weight: ${(props) => props.weight};
  line-height: ${(props) => props.lineHeight}rem;
  ${(props) => props.size === 'h1' ? css`
      font-size: 1.4rem;
      font-weight: 500;
      line-height: 1.6rem;
    ` : props.size === 'h2' ? css`
      font-size: 2rem;
      font-weight: 500;
      line-height: 2.4rem;
      ${theme.breakpoints.mobileMedium(css`
        font-size: 1.7rem;
        line-height: 2rem;
      `)}
    ` : css`
      font-size: 1.4rem;
      line-height: 2.2rem;
    `
  }
`;

// tslint:disable-next-line: variable-name
const LabsCTALink = styled(Button)`
  background-color: #6922ea;
`;

const variants = [
  [
    {
      maxWidth: 17,
      text: [
        {text: 'Kinetic by OpenStax:', size: 'h1'},
        {text: 'Expand Your Learning Potential', size: 'h2'},
      ],
    },
    {
      text: [
        {
          size: 'text',
          text: 'Explore innovative study tools designed to help you discover how you learn best.',
        },
      ],
    },
  ],
  [
    {
      maxWidth: 17,
      text: [
        {text: 'Do you know how you learn best?', size: 'h2'},
      ],
    },
    {
      text: [
        {
          size: 'text',
          text: 'Kinetic by OpenStax offers access to innovative study tools designed' +
                ' to help you maximize your learning potential.',
        },
      ],
    },
  ],
  [
    {
      maxWidth: 25,
      text: [
        {text: 'Kinetic by OpenStax:', size: 'h1'},
        {text: 'A brighter future for digital education starts here.', size: 'h2'},
      ],
    },
    {
      text: [
        {
          size: 'text',
          text: 'See how you can help us create better online learning tools optimized to help everyone grow.',
        },
      ],
    },
  ],
];

// tslint:disable-next-line: variable-name
const LabsCTA = () => {
  const variant = useSelector(kineticBannerVariant);
  const config = typeof variant === 'number' && variants[variant];

  if (!config) {
    return null;
  }

  return (
    <LabsCTAWrapper data-experiment={true}>
      <LabsCTAHeader>
        <Column>
          <LabsLogo src={icon} role='img' alt='' />
        </Column>
        <Column maxWidth={config[0].maxWidth}>
          {config[0].text.map(({text, ...params}, j) => <LabsText {...params} key={j}> {text}</LabsText>)}
        </Column>
      </LabsCTAHeader>
      <Column flex>
        {config[1].text.map(({text, ...params}, j) => <LabsText {...params} key={j}> {text}</LabsText>)}
      </Column>
      <Column last>
        <LabsCTALink
          component={<a href='/kinetic/'>
            <FormattedMessage id='i18n:toolbar:labs-cta:link'>
              {(msg) => msg}
            </FormattedMessage>
          </a>}
          data-analytics-label='kinetic-banner'
          variant='primary'
          size='large'
          target='_blank'
          rel='noopener'
        />
      </Column>
    </LabsCTAWrapper>
  );
};

export default LabsCTA;
