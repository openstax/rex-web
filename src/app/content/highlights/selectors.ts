import { createSelector } from 'reselect';
import * as parentSelectors from '../selectors';
import {
  getHighlightColorFiltersWithContent,
  getHighlightLocationFilters,
  getHighlightLocationFiltersWithContent,
  getSortedSummaryHighlights,
  sectionIsHighlightLocationFitler
} from './utils';
import {
  checkIfHasMoreResults,
  filterCounts,
  getLoadedCountsPerSource,
  getSelectedHighlightsLocationFilters
} from './utils/selectorsUtils';

export const localState = createSelector(
  parentSelectors.localState,
  (parentState) => parentState.highlights
);

export const highlightsLoaded = createSelector(
  localState,
  (state) => state.currentPage.highlights !== null
);

export const highlightsPageId = createSelector(
  localState,
  (state) => state.currentPage.pageId
);

export const highlights = createSelector(
  localState,
  (state) => state.currentPage.highlights || []
);

export const totalCountsPerPage = createSelector(
  localState,
  (state) => state.summary.totalCountsPerPage
);

const totalCountsPerPageOrEmpty = createSelector(
  totalCountsPerPage,
  (totalCounts) => totalCounts || {}
);

export const focused = createSelector(
  localState,
  (state) => state.currentPage.focused
);

export const hasUnsavedHighlight = createSelector(
  localState,
  (state) => state.currentPage.hasUnsavedHighlight
);

export const myHighlightsOpen = createSelector(
  localState,
  (state) => state.summary.open
);

export const summaryIsLoading = createSelector(
  localState,
  (state) => state.summary.loading
);

export const summaryHighlights = createSelector(
  localState,
  (state) => state.summary.highlights
);

export const summaryPagination = createSelector(
  localState,
  (state) => state.summary.pagination
);

export const highlightLocationFilters = createSelector(
  parentSelectors.book,
  getHighlightLocationFilters(sectionIsHighlightLocationFitler)
);

export const orderedSummaryHighlights = createSelector(
  summaryHighlights,
  highlightLocationFilters,
  (highlightsToSort, locationFilters) => {
    return getSortedSummaryHighlights(highlightsToSort, locationFilters);
  }
);

export const highlightLocationFiltersWithContent = createSelector(
  highlightLocationFilters,
  totalCountsPerPageOrEmpty,
  (locationFilters, totalCounts) => getHighlightLocationFiltersWithContent(locationFilters, totalCounts)
);

export const highlightColorFiltersWithContent = createSelector(
  totalCountsPerPageOrEmpty,
  (totalCounts) => getHighlightColorFiltersWithContent(totalCounts)
);

export const loadedCountsPerSource = createSelector(
  summaryHighlights,
  getLoadedCountsPerSource
);

export const summaryFilters = createSelector(
  localState,
  (state) => state.summary.filters
);

const rawSummaryLocationFilters = createSelector(
  summaryFilters,
  (filters) => filters.locationIds
);

const rawSummaryColorFilters = createSelector(
  summaryFilters,
  (filters) => filters.colors
);

export const summaryLocationFilters = createSelector(
  rawSummaryLocationFilters,
  highlightLocationFiltersWithContent,
  (selectedLocations, withContent) =>
    new Set(selectedLocations.filter((locationId) => withContent.has(locationId)))
);

export const summaryColorFilters = createSelector(
  rawSummaryColorFilters,
  highlightColorFiltersWithContent,
  (selectedColors, withContent) =>
    new Set(selectedColors.filter((color) => withContent.has(color)))
);

const selectedHighlightLocationFilters = createSelector(
  highlightLocationFilters,
  summaryLocationFilters,
  getSelectedHighlightsLocationFilters
);

export const filteredCountsPerPage = createSelector(
  totalCountsPerPageOrEmpty,
  selectedHighlightLocationFilters,
  summaryColorFilters,
  filterCounts
);

export const hasMoreResults = createSelector(
  loadedCountsPerSource,
  filteredCountsPerPage,
  summaryPagination,
  checkIfHasMoreResults
);
