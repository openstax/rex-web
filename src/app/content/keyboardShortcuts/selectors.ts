import { createSelector } from 'reselect';
import * as navigationSelectors from '../../navigation/selectors';
import { modalUrlName } from './constants';

export const isKeyboardShortcutsOpen = createSelector(
  navigationSelectors.query, (query) => !!query && query.modal === modalUrlName
);
