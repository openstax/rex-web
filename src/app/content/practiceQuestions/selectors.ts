import { createSelector } from 'reselect';
import * as navigationSelectors from '../../navigation/selectors';
import * as parentSelectors from '../selectors';
import { modalUrlName } from './constants';
import { getPracticeQuestionsLocationFilters } from './utils';

export const localState = createSelector(
  parentSelectors.localState,
  (parentState) => parentState.practiceQuestions
);

export const practiceQuestionsEnabled = createSelector(
  localState,
  (state) => state.isEnabled
);

export const practiceQuestionsSummary = createSelector(
  localState,
  (state) => state.summary
);

export const hasPracticeQuestions = createSelector(
  practiceQuestionsSummary,
  (summary) => Boolean(summary && Object.keys(summary.countsPerSource).length > 0)
);

export const isPracticeQuestionsOpen = createSelector(
  hasPracticeQuestions,
  navigationSelectors.query,
  practiceQuestionsEnabled,
  (hasPQ, query, isEnabled) => hasPQ && query && query.modal === modalUrlName && isEnabled
);

export const practiceQuestionsAreLoading = createSelector(
  localState,
  (state) => state.loading
);

export const practiceQuestionsLocationFilters = createSelector(
  practiceQuestionsSummary,
  parentSelectors.book,
  getPracticeQuestionsLocationFilters
);

export const selectedSection = createSelector(
  localState,
  (state) => state.selectedSection
);

export const questionsCount = createSelector(
  localState,
  (state) => state.questions.length
);

export const currentQuestionIndex = createSelector(
  localState,
  (state) => state.currentQuestionIndex
);

export const question = createSelector(
  localState,
  currentQuestionIndex,
  (state, index) => index === null ? undefined : state.questions[index]
);

export const questionAnswers = createSelector(
  localState,
  (state) => new Map(Object.entries(state.questionAnswers))
);

export const isCurrentQuestionSubmitted = createSelector(
  questionAnswers,
  question,
  (answers, currentQuestion) => Boolean(currentQuestion && answers.has(currentQuestion.uid))
);

export const isFinalQuestion = createSelector(
  questionsCount,
  currentQuestionIndex,
  (count, index) => count - 1 === index
);

export const questionsInProggress = createSelector(
  currentQuestionIndex,
  (index) => index !== null
);

export const hasAnswers = createSelector(
  questionAnswers,
  (answers) => answers.size > 0
);
