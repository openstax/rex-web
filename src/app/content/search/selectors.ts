import { createSelector } from 'reselect';
import * as parentSelectors from '../selectors';
import {
  countTotalHighlights,
  countUniqueKeyTermHighlights,
  getFilteredResults,
  getFormattedSearchResults,
  getNonKeyTermResults,
  getSearchResultsForPage,
  matchKeyTermHit,
} from './utils';

export const localState = createSelector(
  parentSelectors.localState,
  (parentState) => parentState.search
);

export const searchResultsOpen = createSelector(
  localState,
  (state) => !!state.query && state.sidebarOpen
);

const getRawResults = createSelector(
  localState,
  (state) => state.results
);

const filteredResults = createSelector(
  getRawResults,
  (rawResults) => rawResults ? getFilteredResults(rawResults) : null
);

export const hasResults = createSelector(
  filteredResults,
  (selectedResults) => !!selectedResults
);

export const query = createSelector(
  localState,
  (state) => state.query
);

export const selectedResult = createSelector(
  localState,
  (state) => state.selectedResult
);

const nonKTHits = createSelector(
  filteredResults,
  (selectedResults) => selectedResults ? selectedResults.hits.hits.filter((hit) => !matchKeyTermHit(hit)) : null
);

const keyTermHits = createSelector(
  filteredResults,
  (selectedResults) => selectedResults ? selectedResults.hits.hits.filter(matchKeyTermHit) : null
);

export const keyTermHitsInTitle = createSelector(
  keyTermHits,
  (selectedHits) => selectedHits ? selectedHits.filter((hit) => !!hit.highlight.title) : null
);

export const hits = createSelector(
  nonKTHits,
  keyTermHitsInTitle,
  (chapterHits, filteredkeyTermHits) => [...(chapterHits || []), ...(filteredkeyTermHits || [])]
);

export const totalHits = createSelector(
  nonKTHits,
  (hitsOrNull) => hitsOrNull ? countTotalHighlights(hitsOrNull) : null
);

export const totalHitsKeyTerms = createSelector(
  keyTermHitsInTitle,
  (keyTermHitsOrNull) => keyTermHitsOrNull ? countUniqueKeyTermHighlights(keyTermHitsOrNull) : null
);

export const results = createSelector(
  filteredResults,
  parentSelectors.book,
  (selectedResults, book) => selectedResults && book ? getFormattedSearchResults(book.tree, selectedResults) : null
);

const rawNonKTResults = createSelector(
  filteredResults,
  (selectedResults) => selectedResults ? getNonKeyTermResults(selectedResults) : null
);

export const nonKeyTermResults = createSelector(
  rawNonKTResults,
  parentSelectors.book,
  (selectedResults, book) => selectedResults && book ? getFormattedSearchResults(book.tree, selectedResults) : null
);

export const mobileToolbarOpen = createSelector(
  localState,
  (state) => state.mobileToolbarOpen
);

export const currentPageResults = createSelector(
  filteredResults,
  parentSelectors.page,
  (selectedResults, page) => selectedResults && page ? getSearchResultsForPage(page, selectedResults) : []
);

export const userSelectedResult = createSelector(
  localState,
  (state) => state.userSelectedResult
);
