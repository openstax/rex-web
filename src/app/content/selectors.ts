import { createSelector } from 'reselect';
import * as parentSelectors from '../selectors';
import { defaultTheme } from './components/constants';
import { hasOSWebData } from './guards';
import { getIdFromPageParam } from './utils';
import {
  findArchiveTreeNodeByPageParam,
  prevNextBookPage,
} from './utils/archiveTreeUtils';

export const localState = createSelector(
  parentSelectors.localState,
  (parentState) => parentState.content
);

export const tocOpen = createSelector(
  localState,
  (state) => !state.search.query && state.tocOpen
);

export const book = createSelector(
  localState,
  (state) => state.book
);

export const buyPrintConfig = createSelector(
  localState,
  (state) => state.buyPrint
);

export const bookTheme = createSelector(
  book,
  (currentBook) => hasOSWebData(currentBook) ? currentBook.theme : defaultTheme
);

export const showNudgeStudyTools = createSelector(
  localState,
  (state) => state.showNudgeStudyTools
);

export const contentReferences = createSelector(
  localState,
  (state) => state.references
);

export const page = createSelector(
  localState,
  (state) => state.page
);

export const loading = createSelector(
  localState,
  (state) => state.loading
);

export const loadingBook = createSelector(
  loading,
  (slugs) => slugs.book
);

export const loadingPage = createSelector(
  loading,
  (slugs) => slugs.page
);

export const contentParams = createSelector(
  localState,
  (state) => state.params
);

export const pageParam = createSelector(
  contentParams,
  (params) => params ? params.page : null
);

export const pageNotFound = createSelector(
  localState,
  pageParam,
  (state, param) => getIdFromPageParam(param) === state.pageNotFoundId
);

export const pageNode = createSelector(
  book,
  pageParam,
  (selectedBook, selectedPageParam) => selectedBook && selectedPageParam
    ? findArchiveTreeNodeByPageParam(selectedBook.tree, selectedPageParam)
    : undefined
);

export const bookAndPage = createSelector(
  book,
  page,
  (selectedBook, selectedPage) => ({book: selectedBook, page: selectedPage})
);

export const prevNextPage = createSelector(
  book,
  page,
  (selectedBook, selectedPage) => selectedBook && selectedPage
    ? prevNextBookPage(selectedBook, selectedPage.id)
    : null
);
