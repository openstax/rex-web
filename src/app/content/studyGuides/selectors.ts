import { createSelector } from 'reselect';
import { getHighlightColorFiltersWithContent, getHighlightLocationFilterForPage } from '../highlights/utils';
import {
  getHighlightLocationFilters,
  getHighlightLocationFiltersWithContent,
  getSortedSummaryHighlights
} from '../highlights/utils';
import {
  checkIfHasMoreResults,
  filterCounts,
  getLoadedCountsPerSource,
  getSelectedHighlightsLocationFilters,
} from '../highlights/utils/selectorsUtils';
import * as parentSelectors from '../selectors';
import { archiveTreeSectionIsChapter } from '../utils/archiveTreeUtils';

export const localState = createSelector(
  parentSelectors.localState,
  (parentState) => parentState.studyGuides
);

export const studyGuidesEnabled = createSelector(
  localState,
  (state) => state.isEnabled
);

export const studyGuidesSummary = createSelector(
  localState,
  (state) => state.summary
);

export const hasStudyGuides = createSelector(
  studyGuidesSummary,
  (summary) => summary.totalCountsPerPage
    && Object.keys(summary.totalCountsPerPage).length > 0
);

export const totalCountsPerPage = createSelector(
  studyGuidesSummary,
  (summary) => summary.totalCountsPerPage
);

export const totalCountsPerPageOrEmpty = createSelector(
  studyGuidesSummary,
  (summary) => summary.totalCountsPerPage || {}
);

export const studyGuidesOpen = createSelector(
  studyGuidesSummary,
  (summary) => summary.open
);

export const summaryIsLoading = createSelector(
  studyGuidesSummary,
  (summary) => summary.loading
);

export const summaryStudyGuidesPagination = createSelector(
  studyGuidesSummary,
  (summary) => summary.pagination
);

export const studyGuidesLocationFilters = createSelector(
  parentSelectors.book,
  getHighlightLocationFilters(archiveTreeSectionIsChapter)
);

export const summaryStudyGuides = createSelector(
  studyGuidesSummary,
  (summary) => summary.studyGuides
);

export const orderedSummaryStudyGuides = createSelector(
  summaryStudyGuides,
  studyGuidesLocationFilters,
  getSortedSummaryHighlights
);

export const loadedCountsPerSource = createSelector(
  summaryStudyGuides,
  getLoadedCountsPerSource
);

export const summaryFilters = createSelector(
  localState,
  (state) => state.summary.filters
);

const rawSummaryLocationFilters = createSelector(
  summaryFilters,
  (filters) => filters.locationIds
);

export const highlightColorFiltersWithContent = createSelector(
  totalCountsPerPageOrEmpty,
  getHighlightColorFiltersWithContent
);

export const summaryColorFilters = createSelector(
  summaryFilters,
  highlightColorFiltersWithContent,
  (filters, withContent) => new Set(filters.colors.filter((color) => withContent.has(color)))
);

export const studyGuidesLocationFiltersWithContent = createSelector(
  studyGuidesLocationFilters,
  totalCountsPerPageOrEmpty,
  getHighlightLocationFiltersWithContent
);

export const defaultLocationFilter = createSelector(
  studyGuidesLocationFilters,
  parentSelectors.page,
  (filters, currentPage) => currentPage && getHighlightLocationFilterForPage(filters, currentPage)
);

export const summaryLocationFilters = createSelector(
  rawSummaryLocationFilters,
  (selectedLocations) => new Set(selectedLocations)
);

const selectedStudyGuidesLocationFilters = createSelector(
  studyGuidesLocationFilters,
  summaryLocationFilters,
  getSelectedHighlightsLocationFilters
);

export const filteredCountsPerPage = createSelector(
  totalCountsPerPageOrEmpty,
  selectedStudyGuidesLocationFilters,
  summaryColorFilters,
  filterCounts
);

export const hasMoreResults = createSelector(
  loadedCountsPerSource,
  filteredCountsPerPage,
  summaryStudyGuidesPagination,
  checkIfHasMoreResults
);
