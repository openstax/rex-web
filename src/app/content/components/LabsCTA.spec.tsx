import React from 'react';
import renderer from 'react-test-renderer';
import TestContainer from '../../../test/TestContainer';
import * as selectors from '../../featureFlags/selectors';
import LabsCTA from './LabsCTA';

describe('PrevNextBar', () => {
  let variantSelectorSpy: jest.SpyInstance;

  beforeEach(() => {
    variantSelectorSpy = jest.spyOn(selectors, 'kineticBannerVariant');
  });

  const render = () => <TestContainer>
    <LabsCTA />
  </TestContainer>;

  it('renders `null` when selector is false', () => {
    variantSelectorSpy.mockReturnValue(false);
    const component = renderer.create(render());

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders first variant', () => {
    variantSelectorSpy.mockReturnValue(0);
    const component = renderer.create(render());

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders second variant', () => {
    variantSelectorSpy.mockReturnValue(1);
    const component = renderer.create(render());

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders third variant', () => {
    variantSelectorSpy.mockReturnValue(2);
    const component = renderer.create(render());

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
