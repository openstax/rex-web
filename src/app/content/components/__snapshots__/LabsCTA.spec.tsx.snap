// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrevNextBar renders \`null\` when selector is false 1`] = `null`;

exports[`PrevNextBar renders first variant 1`] = `
<div
  className="LabsCTA__LabsCTAWrapper-sc-117qi5r-1 figEZj"
  data-experiment={true}
>
  <div
    className="LabsCTA__LabsCTAHeader-sc-117qi5r-3 gvQZQT"
  >
    <div
      className="LabsCTA__Column-sc-117qi5r-2 cRAhAC"
    >
      <img
        alt=""
        className="LabsCTA__LabsLogo-sc-117qi5r-0 LUvlV"
        role="img"
        src="kinetic.svg"
      />
    </div>
    <div
      className="LabsCTA__Column-sc-117qi5r-2 vXuFz"
    >
      <div
        className="LabsCTA__LabsText-sc-117qi5r-4 khUpZV"
        size="h1"
      >
         
        Kinetic by OpenStax:
      </div>
      <div
        className="LabsCTA__LabsText-sc-117qi5r-4 fnabEM"
        size="h2"
      >
         
        Expand Your Learning Potential
      </div>
    </div>
  </div>
  <div
    className="LabsCTA__Column-sc-117qi5r-2 ciJCWM"
  >
    <div
      className="LabsCTA__LabsText-sc-117qi5r-4 hgTBzT"
      size="text"
    >
       
      Explore innovative study tools designed to help you discover how you learn best.
    </div>
  </div>
  <div
    className="LabsCTA__Column-sc-117qi5r-2 hKDNCR"
  >
    <a
      className="Button-ayg7nk-0 LabsCTA__LabsCTALink-sc-117qi5r-5 dzloHR"
      data-analytics-label="kinetic-banner"
      href="/kinetic/"
      rel="noopener"
      target="_blank"
    >
      Explore Kinetic
    </a>
  </div>
</div>
`;

exports[`PrevNextBar renders second variant 1`] = `
<div
  className="LabsCTA__LabsCTAWrapper-sc-117qi5r-1 figEZj"
  data-experiment={true}
>
  <div
    className="LabsCTA__LabsCTAHeader-sc-117qi5r-3 gvQZQT"
  >
    <div
      className="LabsCTA__Column-sc-117qi5r-2 cRAhAC"
    >
      <img
        alt=""
        className="LabsCTA__LabsLogo-sc-117qi5r-0 LUvlV"
        role="img"
        src="kinetic.svg"
      />
    </div>
    <div
      className="LabsCTA__Column-sc-117qi5r-2 vXuFz"
    >
      <div
        className="LabsCTA__LabsText-sc-117qi5r-4 fnabEM"
        size="h2"
      >
         
        Do you know how you learn best?
      </div>
    </div>
  </div>
  <div
    className="LabsCTA__Column-sc-117qi5r-2 ciJCWM"
  >
    <div
      className="LabsCTA__LabsText-sc-117qi5r-4 hgTBzT"
      size="text"
    >
       
      Kinetic by OpenStax offers access to innovative study tools designed to help you maximize your learning potential.
    </div>
  </div>
  <div
    className="LabsCTA__Column-sc-117qi5r-2 hKDNCR"
  >
    <a
      className="Button-ayg7nk-0 LabsCTA__LabsCTALink-sc-117qi5r-5 dzloHR"
      data-analytics-label="kinetic-banner"
      href="/kinetic/"
      rel="noopener"
      target="_blank"
    >
      Explore Kinetic
    </a>
  </div>
</div>
`;

exports[`PrevNextBar renders third variant 1`] = `
<div
  className="LabsCTA__LabsCTAWrapper-sc-117qi5r-1 figEZj"
  data-experiment={true}
>
  <div
    className="LabsCTA__LabsCTAHeader-sc-117qi5r-3 gvQZQT"
  >
    <div
      className="LabsCTA__Column-sc-117qi5r-2 cRAhAC"
    >
      <img
        alt=""
        className="LabsCTA__LabsLogo-sc-117qi5r-0 LUvlV"
        role="img"
        src="kinetic.svg"
      />
    </div>
    <div
      className="LabsCTA__Column-sc-117qi5r-2 izlHRA"
    >
      <div
        className="LabsCTA__LabsText-sc-117qi5r-4 khUpZV"
        size="h1"
      >
         
        Kinetic by OpenStax:
      </div>
      <div
        className="LabsCTA__LabsText-sc-117qi5r-4 fnabEM"
        size="h2"
      >
         
        A brighter future for digital education starts here.
      </div>
    </div>
  </div>
  <div
    className="LabsCTA__Column-sc-117qi5r-2 ciJCWM"
  >
    <div
      className="LabsCTA__LabsText-sc-117qi5r-4 hgTBzT"
      size="text"
    >
       
      See how you can help us create better online learning tools optimized to help everyone grow.
    </div>
  </div>
  <div
    className="LabsCTA__Column-sc-117qi5r-2 hKDNCR"
  >
    <a
      className="Button-ayg7nk-0 LabsCTA__LabsCTALink-sc-117qi5r-5 dzloHR"
      data-analytics-label="kinetic-banner"
      href="/kinetic/"
      rel="noopener"
      target="_blank"
    >
      Explore Kinetic
    </a>
  </div>
</div>
`;
